hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterRpc.java

```diff
@@ -1224,6 +1224,17 @@ public void testProxyConcatFile() throws Exception {
     String badPath = "/unknownlocation/unknowndir";
     compareResponses(routerProtocol, nnProtocol, m,
         new Object[] {badPath, new String[] {routerFile}});
+
+    // Test when concat trg is a empty file
```

Comment:
When there is a empty source file, it will throw exception in namenode.    This behaiver is as same as with dfsrouter.
And  when trg is a empty file, it is diffrent . Without dfsrouter ,it success. And with dfsrouter, it will throw Exception.
So I think  there is no need to check the empty source file. Or  implement it in another PR.

Reply:
Do the namenode and rbf throw the same Exception?

Maybe RBF throws NPE, but NN throws `org.apache.hadoop.HadoopIllegalArgumentException`.

Reply:
When srclist has a empty file ,both namenode and dfsrouter will throw the same IOException 
https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirConcatOp.java#L153-L155

Reply:
Can you modify the UT to cover the case that one or more source files are empty?

Reply:
Add test for a empty  src file